lib/puppet/provider/netapp_ldap_client/cmode.rb

```diff
@@ -0,0 +1,146 @@
+        :ad_domain                  => ldapclient.child_get_string('ad-domain'),
+        :allow_ssl                  => ldapclient.child_get_string('allow-ssl'),
+        :base_dn                    => ldapclient.child_get_string('base-dn'),
+        :base_scope                 => ldapclient.child_get_string('base-scope'),
+        :bind_as_cifs_server        => ldapclient.child_get_string('bind-as-cifs-server'),
+        :bind_dn                    => ldapclient.child_get_string('bind-dn'),
+        :bind_password              => ldapclient.child_get_string('bind-password'),
+        :group_dn                   => ldapclient.child_get_string('group-dn'),
+        :group_scope                => ldapclient.child_get_string('group-scope'),
+        :is_netgroup_byhost_enabled => ldapclient.child_get_string('is-netgroup-byhost-enabled '),
+        :min_bind_level             => ldapclient.child_get_string('min-bind-level'),
+        :netgroup_byhost_dn         => ldapclient.child_get_string('netgroup-byhost-dn'),
+        :netgroup_byhost_scope      => ldapclient.child_get_string('netgroup-byhost-scope'),
+        :netgroup_dn                => ldapclient.child_get_string('netgroup-dn'),
+        :netgroup_scope             => ldapclient.child_get_string('netgroup-scope'),
+        :preferred_ad_servers       => preffered_servers,
+        :query_timeout              => ldapclient.child_get_string('query-timeout'),
+        :schema                     => ldapclient.child_get_string('schema'),
+        :servers                    => servers,
+        :tcp_port                   => ldapclient.child_get_string('tcp-port'),
+        :use_start_tls              => ldapclient.child_get_string('use-start-tls'),
+        :user_dn                    => ldapclient.child_get_string('user-dn'),
+        :user_scope                 => ldapclient.child_get_string('user-scope'),
+        :ensure                     => :present,
+      }
+      ldapclients << new(ldapclient_hash)
+    end
+    ldapclients
+  end
+
+  def self.prefetch(resources)
+    instances.each do |prov|
+      if resource = resources[prov.name]
+        resource.provider = prov
+      end
+    end
+  end
+
+  def flush
+    case @property_hash[:ensure]
+    when :absent
+      ldapclientdestroy('ldap-client-config',resource[:name])
+    when :present
+      ldapclientmodify(*get_args('modify'))
+    end
+  end
+
+  def create
+    ldapclientcreate(*get_args('create'))
+  end
+
+  def destroy
+    @property_hash[:ensure] = :absent
+  end
+
+  def exists?
+    @property_hash[:ensure] == :present
+  end
+
+  def get_args (method)
+    args = NaElement.new("ldap-client-#{method}")
+    args.child_add_string('ldap-client-config', @resource[:name])
+    args.child_add_string('ad-domain', @resource[:ad_domain])
+    args.child_add_string('allow-ssl', @resource[:allow_ssl]) unless @resource[:allow_ssl].nil?
+    args.child_add_string('base-dn', @resource[:base_dn])
```

Comment:
We probably want `unless @resource[:whatever].nil?` on all values that are not always available.